provider/manual/provider.go

```diff
@@ -85,7 +85,13 @@ func pingMachine(endpoint string) error {
 	cmd.Stderr = &buf
 	if err := cmd.Run(); err != nil {
 		if buf.Len() > 0 {
-			return errors.New(buf.String())
+			received := buf.String()
+			if strings.HasPrefix(received, "Permission denied") {
```

Comment:
I'm pretty sure this is not going to work with the x/crypto/ssh SSH client implementation. Can you please test that?

I'm wondering if we're not better off just modifying the Ping method to take a context, passing in stdin/stdout and call the ensureBootstrapUbuntuUser function like we do in PrepareForBootstrap. That way you'll be prompted for your username/password as needed when adding the cloud interactively.